pkg/asset/installconfig/aws/validation.go

```diff
@@ -255,6 +255,41 @@ func validateMachinePool(ctx context.Context, meta *Metadata, fldPath *field.Pat
 			allErrs = append(allErrs, field.Invalid(fldPath.Child("type"), pool.InstanceType, errMsg))
 		}
 	}
+
+	if len(pool.AdditionalSecurityGroupIDs) > 0 {
+		allErrs = append(allErrs, validateSecurityGroupIDs(ctx, meta, fldPath.Child("additionalSecurityGroupIDs"), platform, pool)...)
+	}
+
+	return allErrs
+}
+
+func validateSecurityGroupIDs(ctx context.Context, meta *Metadata, fldPath *field.Path, platform *awstypes.Platform, pool *awstypes.MachinePool) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	vpcs, err := GetVPCsFromSecurityGroups(ctx, meta.session, pool.AdditionalSecurityGroupIDs, platform.Region)
+	if err != nil {
+		return append(allErrs, field.Invalid(fldPath, pool.AdditionalSecurityGroupIDs, err.Error()))
+	}
+
+	// If there are more than 1 VPC associated with the group of security groups fail because
+	// the install requires a single VPC. If zero are found this indicates that there is no
+	// VPC that is common to the security groups and the subnets.
+	if len(vpcs) != 1 {
+		errMsg := fmt.Sprintf("exactly 1 vpc expected; found %d instead", len(vpcs))
+		return append(allErrs, field.Invalid(fldPath, vpcs, errMsg))
+	}
+
+	vpcFromSecurityGroups := vpcs[0]
+	vpc, err := meta.VPC(ctx)
+	if err != nil {
+		errMsg := fmt.Sprintf("could not determine if securityGroups VPC matches the cluster's: %s", err.Error())
+		return append(allErrs, field.Invalid(fldPath, vpcFromSecurityGroups, errMsg))
+	}
+	if vpc != vpcFromSecurityGroups {
+		errMsg := fmt.Sprintf("vpcs do not match %s, %s", vpcFromSecurityGroups, vpc)
```

Comment:
on a similar note to my other comment, this error message could be a little more clear. I think the same suggestion as the other (it seems like these validations could be combined): "sg x is associated with vpc y not the provided vpc z"